### PR TITLE
Add example with high number of morph targets

### PR DESCRIPTION
**Pull request by looeee**
Added a new example to showcase simultaneous animation of a large number of morph targets. 

[Live link](https://raw.githack.com/mrdoob/three.js/2a54e4c85dfc667eb2769606f5b4982979f3a6b8/examples/webgl_morphtargets_face.html) (**note**: normals incorrect pending build with #22516)

main1

The model was kindly provided by the creator of the iPhone [Face Cap](https://www.bannaflak.com/face-cap/) app. 

Possibly this example can replace [webgl_morphtargets_horse](https://threejs.org/examples/#webgl_morphtargets_horse) as they serve a similar function.

### CODE REVIEW

**Comment by looeee** [comment]
I've created this example as draft because the model is currently quite large (3.9mb). I was able to shrink it down using gltfpack to just 325kb. 

[facecap-gltfpack.zip](https://github.com/mrdoob/three.js/files/7135931/facecap-gltfpack.zip)

However when I try to load it this is the result (animation plays correctly, lighting or normals are incorrect):

mess

This only occurs when I set `scene.environment = texture`. If I remove that line and instead add a light, it looks fine. 

``` js
scene.add(new THREE.HemisphereLight());
// scene.environment = texture;
```

light only

However if I drop back to r131 and re-able the environment it looks fine (except now the animation can only use 8 morphtargets)

r131

**Comment by Mugen87** [comment]
I would remove the environment map from the scene and use the more neutral `RoomEnvironment` instead. Then it looks more like the screenshot from https://github.com/mrdoob/three.js/pull/22293#issuecomment-915019398.

**Comment by Mugen87** [comment]
> If I remove that line and instead add a light, it looks fine.

Have you just tested with a hemisphere light? If so, try it again with an additional directional or point light.

File: examples/files.json [hunk 1]
```diff
@@ -1,409 +1,399 @@
- {
```
Comment (Mugen87) [inline examples/files.json:1]: Formatting.

**Comment by Mugen87** [comment]
Okay, the issue happens because `WebGLMorphtargets` assumes that the morph attribute data are represented as `Float32`. When using gltf-pack, the attribute data are compressed to `Int16Array`. Reading these data to setup the data texture does not work yet.

**Comment by Mugen87** [comment]
Since the data texture needs to be of type `FloatType`, applying the attribute compression is not only limited useful in the first place because the the data needs to be converted back to float.

Is it possible to setup gltf-pack that morph data are not compressed?

**Comment by Mugen87** [comment]
@mrdoob I can fix this if you want by doing this in `WebGLMorphtargets`:
```js
if ( morphNormal.normalized === true ) {

    const denominator = getMaxIntegerValue( morphNormal );

    morph.x /= denominator;
    morph.y /= denominator;
    morph.z /= denominator;

}
```
`getMaxIntegerValue()` would be a new function that checks the attribute array for `Int8Array`, `Int16Array` or `Int32Array` and then determines the respective max value. This value will be used to denormlize the morph normal.

**Comment by donmccurdy** [comment]
This model could be an option as well:

https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/MorphStressTest

I think it would be helpful to support normalized morph attributes, but am happy to help create a version of either model that doesn't include them if needed.

**Comment by spiraloid** [comment]
+ 1 for using this model.  the shapes it has are based on the Animoji FACS (based on my Gollum FACS from LOTR)

I'd suggest skinning to some bones (w 4 bone linear skinning)  so that the skinning, morph targets and normals are all on the same gpu loop.   

all three are required for character animation.   doing just one in isolation makes no sense.

Protip:   the vertex positions and normal precision can be done as INTS (i.e. pixels) as long as they are normalized into the bounding dimension of the largest vertex delta in all the morphs.  the visual fidelity loss us un-noticeable  ;)

existence proof:
https://youtu.be/-cSFPIwMEq4?t=11

File: examples/files.json (hunk 1 above)
Comment (looeee) [reply examples/files.json:1]: Should be fixed I think.

**Comment by looeee** [comment]
> I would remove the environment map from the scene and use the more neutral `RoomEnvironment` instead. Then it looks more like the screenshot from [#22293 (comment)](https://github.com/mrdoob/three.js/pull/22293#issuecomment-915019398).

Done, I've updated the live link.

**Comment by looeee** [comment]
> Is it possible to configure gltf-pack such that morph data are not compressed?

I don't think so. I tried making an uncompressed file by running `gltfpack -i facecap.glb -o out.glb` but the result is the same (filesize around 2.5mb). 

@donmccurdy I did also try gltf-transform, both draco and meshopt. Draco barely reduced the filesize and meshopt completed messed up the model. I ran both commands with default settings, but investigate any further yet.

**Comment by mrdoob** [comment]
@spiraloid 

> happy to skin up a neck to that mesh if it helps keep the test case legit. lemme know.

That'd be super useful!

**Comment by Mugen87** [comment]
> I don't think so. I tried making an uncompressed file by running gltfpack -i facecap.glb -o out.glb but the result is the same (filesize around 2.5mb).

@looeee Never mind! With #22516 it's possible to use the model from gltf-pack.

**Comment by donmccurdy** [comment]
> I did also try gltf-transform, both draco and meshopt. Draco barely reduced the filesize and meshopt completed messed up the model. I ran both commands with default settings, but didn't investigate any further yet.

Ok thanks! Draco won't do anything with morph targets, but I would've expected meshopt to work. Filed a bug at https://github.com/donmccurdy/glTF-Transform/issues/352.

Glad to see https://github.com/mrdoob/three.js/pull/22516!

**Comment by spiraloid** [comment]
draco not compressing the targets could be heavy depending how many verts are moving.  storing the morph target/normals as a png could be a smart option.  each 3 pixels being the unitized position of each vert in an index that gets expanded back to float on draw.  that way draco image compression could compress the morph targets….  an idea.

**Comment by donmccurdy** [comment]
Draco cannot compress images, though. I think meshopt compression is the right choice here (this is what `gltfpack` uses). Custom compression schemes (and custom data pipelines to produce them...) can be very effective for particular applications, but I don't think we should be pointing people that way in our examples.

**Comment by looeee** [comment]
OK, I've switched to the gltfpack compressed model and marked this PR as ready for review.

**Comment by looeee** [comment]
@spiraloid giving the model a neck would be great 👍 I think I have the original FBX if that's easier for you to work from.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**